package/MDAnalysis/analysis/align.py

```diff
@@ -504,6 +511,15 @@ def alignto(mobile, reference, select=None, weights=None,
     if subselection is None:
         # mobile_atoms is Universe
         mobile_atoms = mobile.universe.atoms
+        if mobile.universe == reference.universe:
+            # raise a warning when passing atom groups from same Universe
+            wmsg = ("The reference and mobile selections are atom groups "
+                    "from the same universe, the reference selection will "
+                    "be transformed. If it is not the behavior you intend, "
+                    "try setting the subselection argument to 'all'. "
+                    "`subselection = 'all' is deprecated in 2.1.0 but it "
+                    "will become the default behavior in version 3.0.0.")
```

Comment:
This is incorrect, `subselection='all'` is not deprecated. What you should be saying here is that the default behaviour will change to that of `subselection='all'` in version 3.0.0.